src/handlers/comment/handlers/assign/generate-assignment-comment.ts

```diff
@@ -0,0 +1,43 @@
+import Runtime from "../../../../bindings/bot-runtime";
+import { Context } from "../../../../types/context";
+import { Payload } from "../../../../types/payload";
+
+const options: Intl.DateTimeFormatOptions = {
+  weekday: "short",
+  month: "short",
+  day: "numeric",
+  hour: "numeric",
+  minute: "numeric",
+  timeZone: "UTC",
+  timeZoneName: "short",
+};
+
+export async function generateAssignmentComment(context: Context, payload: Payload, duration: number | null = null) {
+  const runtime = Runtime.getState();
+  const startTime = new Date().getTime();
+  let endTime: null | Date = null;
+  let deadline: null | string = null;
+  if (duration) {
+    endTime = new Date(startTime + duration * 1000);
+    deadline = endTime.toLocaleString("en-US", options);
+  }
+
+  const issueCreationTime = payload.issue?.created_at;
+  if (!issueCreationTime) {
+    throw context.logger.fatal("Issue creation time is not defined");
+  }
+
+  return {
+    daysElapsedSinceTaskCreation: Math.floor((startTime - new Date(issueCreationTime).getTime()) / 1000 / 60 / 60 / 24),
+    deadline,
+    registeredWallet:
+      (await runtime.adapters.supabase.wallet.getAddress(payload.sender.id)) ||
+      "Please set your wallet address to use `/wallet 0x0000...0000`",
```

Comment:
We should throw an error to force them to set a wallet address before self assignment.